packaging/rpm/foundationdb.service

```diff
@@ -7,7 +7,9 @@ Wants=network-online.target
 Type=forking
 PIDFile=/var/run/fdbmonitor.pid
 ExecStart=/usr/lib/foundationdb/fdbmonitor --conffile /etc/foundationdb/foundationdb.conf --lockfile /var/run/fdbmonitor.pid --daemonize
-KillMode=process
+KillMode=mixed
+Restart=on-failure
+RestartSec=5s
```

Comment:
If my understanding is correct, there are two different things happening here:

1. Changing the KillMode ensures the children get killed along with fdbmonitor
2. Changing the restart params allows fdbmonitor to automatically restart

I'm happy with the first part of this change. It seems reasonable to me that the spawned processes not have a lifetime that extends beyond fdbmonitor.

For the second change, I have a few thoughts. Currently, I think only the macOS service restarts fdbmonitor automatically, whereas in Linux and Windows we do not. Neither macOS or Windows are designed around running in production, but for Linux in particular I wonder if there's a benefit to avoiding restarting on too tight a loop if fdbmonitor repeatedly dies. Otherwise, we could end up having processes repeatedly join the cluster and die, resulting in negative effects on the cluster. Our foundationdb.conf file has a backoff mechanism for server processes, but that gets defeated if fdbmonitor is what's restarting.

My inclination is to say that this is ok, but we should do a few things:

1. Increase the default restart delay to something longer, perhaps 60 seconds?
2. Document this behavior and how the parameters can be adjusted if needed (modifying a conf file in `/etc`?).
3. Create issues to implement this behavior at least in the deb packages and maybe Windows too.

Reply:
> What file should I mention it in?

I guess the best place to actually put the description of the behavior and how to change is in its own section in the configuration document:

https://github.com/apple/foundationdb/blob/master/documentation/sphinx/source/configuration.rst

I think it probably makes sense to also mention restart behavior in the "Starting and stopping" section, linking to the above section:

https://apple.github.io/foundationdb/administration.html#starting-and-stopping
(located in https://github.com/apple/foundationdb/blob/master/documentation/sphinx/source/administration.rst)

It may also be a good idea to note in the foundationdb.conf "[general]" section where we describe process restart behavior that the outcome of fdbmonitor dying is platform dependent, again linking to our new docs:

https://apple.github.io/foundationdb/configuration.html#general-section
(located in https://github.com/apple/foundationdb/blob/master/documentation/sphinx/source/configuration.rst)

Reply:
Done. https://github.com/apple/foundationdb/pull/3841/commits/da96ae7c1eff8d12adad8f1384751ac012ee1454